### PR TITLE
Workaround for landau4d-openmp problem

### PR DESCRIPTION
**Pull request by blegouix**
https://github.com/ginkgo-project/ginkgo/discussions/1561

### CODE REVIEW

**Comment by tpadioleau** [comment]
So it means we lose parallelism ?

**Comment by tpadioleau** [comment]
I agree to merge it once we have a small reproducer

File: include/ddc/kernels/splines/matrix_sparse.hpp [hunk 1]
```diff
@@ -164,7 +164,13 @@ class Matrix_Sparse : public Matrix
                 = gko::solver::Bicgstab<double>::build()
                           .with_preconditioner(preconditioner)
                           .with_criteria(residual_criterion, iterations_criterion)
+#ifdef KOKKOS_ENABLE_OPENMP
+                          .on(std::is_same_v<ExecSpace, Kokkos::OpenMP>
+                                      ? create_gko_exec<Kokkos::Serial>()
+                                      : gko_exec); // Workaround because unexpected behavior is observed in gslx::Landau4D with dx=0 in advection_x in OpenMP, even with OMP_NUM_THREADS=1
```
Comment (blegouix) [inline include/ddc/kernels/splines/matrix_sparse.hpp:170]:
```suggestion
                                      : gko_exec); // Workaround related to #332
```

File: include/ddc/kernels/splines/matrix_sparse.hpp [hunk 2]
```diff
@@ -74,12 +74,12 @@ unsigned int default_preconditionner_max_block_size() noexcept
 #endif
 #ifdef KOKKOS_ENABLE_OPENMP
     if (std::is_same_v<ExecSpace, Kokkos::OpenMP>) {
-        return 32u;
+        return 1u; // Related to #1563 issue
```
Comment (blegouix) [inline include/ddc/kernels/splines/matrix_sparse.hpp:77]:
```suggestion
        return 1u; // Related to Ginkgo#1563 issue
```

**Comment by tpadioleau** [comment]
Shall we delete the branch ?

**Pull request closed without merging**